File: back-end/src/main/java/acc/br/shopbank/domain/model/CheckingAccount.java
package acc.br.shopbank.domain.model;

import acc.br.shopbank.domain.enums.AccountType;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "checking_accounts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CheckingAccount {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "bank_id")
    private Bank bank;

    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @ManyToOne
    @JoinColumn(name = "store_id")
    private Store store;

    private String agency;

    private String number;

    private String digit;

    @Column(nullable = false, precision = 15, scale = 2)
    private BigDecimal balance;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AccountType type;

    private Boolean active;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    public void prePersist() {
        balance = balance == null ? BigDecimal.ZERO : balance;
        active = active == null ? Boolean.TRUE : active;
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    public void preUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: mobile/src/types/checkingAccount.ts
export type AccountType = 'CUSTOMER' | 'STORE' | 'MARKETPLACE';

export type TransactionType =
  | 'CREDIT'
  | 'DEBIT'
  | 'REFUND'
  | 'DEPOSIT'
  | 'WITHDRAWAL';

export interface CheckingAccount {
  id: string;
  bankId: string;
  customerId?: string;
  storeId?: string;
  agency: string;
  number: string;
  digit: string;
  balance: number;
  type: AccountType;
  active: boolean;
}

export interface AccountTransaction {
  id: string;
  checkingAccountId: string;
  orderId?: string;
  paymentId?: string;
  type: TransactionType;
  amount: number;
  description?: string;
  createdAt: string;
}

export interface AmountRequest {
  amount: number;
  description?: string;
}

export interface TransactionFilter {
  from?: string;
  to?: string;
  type?: TransactionType;
}
